player/lua/console.lua

```diff
@@ -1748,6 +1755,44 @@ local function undefine_key_bindings()
     key_bindings = {}
 end
 
+local function read_history()
+    if opts.persist_history == false or history[1] then
+        return
+    end
+
+    local history_file = io.open(mp.command_native({'expand-path', opts.history_path}))
+
+    if history_file == nil then
+        return
+    end
+
+    if opts.history_dedup then
+        local unfiltered_history = {}
+        for command in history_file:lines() do
+            unfiltered_history[#unfiltered_history + 1] = command
+        end
+
+        local history_map = {}
+        for i = #unfiltered_history, 1, -1 do
+            local command = unfiltered_history[i]
+            if not history_map[command] then
```

Comment:
I wouldn't do global de-duplication to preserve order of commands executed.

So if we execute commands
```
a
a
a
b
a
a
b
b
a
```
it should be
```
a
b
a
b
a
```
instead of 
```
a
b
```

What do you think?

Reply:
In my own console history implementations I do global deduplication, and it did not happen even once that I wished that identical copies of the same command were retained.

I'm not saying there can't be use cases for keeping duplicates, but I can't think of one.

Reply:
Why? Deduplication should be global to not have duplicate result in history search.

Reply:
> Why? Deduplication should be global to not have duplicate results in history search.

If you go UP with an arrow it normally loops the history without consecutive duplicates. Same with ctrl+r search.

This is how zsh works for example. So I'm asking if mpv should be different that established implementations. For example if you want to repeat sequence of commands it is possible in such design, if it's globally the information of the sequence of operations is lost.

Reply:
That's definitely not how my zsh works. Again, you need to set `HIST_IGNORE_ALL_DUPS`, and fish which has good default does this by default, while zsh has terrible defaults that we shouldn't imitate, else we should e.g. disable command-specific completions by default.